File: cef-web/src/main/java/gov/epa/cef/web/service/FacilitySiteService.java
package gov.epa.cef.web.service;

import gov.epa.cef.web.domain.FacilitySite;
import gov.epa.cef.web.service.dto.FacilityNAICSDto;
import gov.epa.cef.web.service.dto.FacilitySiteDto;

public interface FacilitySiteService {

    /**
     * Create a new facilitySite
     * @param facilitySite
     * @return
     */
	FacilitySiteDto create(FacilitySite facilitySite);

    /**
     * Find facility by ID
     * @param id
     * @return
     */
    FacilitySiteDto findById(Long id);

    /**
     * Retrieve facility by emissions report
     * @param emissionsReportId
     * @return
     */
    FacilitySiteDto findByReportId(Long emissionsReportId);

    /**
     * Update facility information
     * @param dto
     * @return
     */
    FacilitySiteDto update(FacilitySiteDto dto);

    /**
     * Create Facility NAICS
     * @param dto
     */
    FacilityNAICSDto createNaics(FacilityNAICSDto dto);

    /**
     * Update existing facility NAICS
     * @param dto
     * @return
     */
    FacilityNAICSDto updateNaics(FacilityNAICSDto dto);

    /**
     * Delete Facility NAICS by id
     * @param facilityNaicsId
     */
    void deleteFacilityNaics(Long facilityNaicsId);


    /**
     * Transform from DTO to new instance FacilitySite
     * @return
     */
    FacilitySite transform(FacilitySiteDto dto);
}

File: cef-web/src/main/webapp/src/app/core/services/facility-site.service.ts
import {CefFacility} from 'src/app/shared/models/cef-facility';
import {FacilitySite} from 'src/app/shared/models/facility-site';
import {HttpClient} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import { FacilityNaicsCode } from 'src/app/shared/models/facility-naics-code';


@Injectable({
    providedIn: 'root'
})
export class FacilitySiteService {

    private baseUrl = 'api/facilitySite';  // URL to web api

    constructor(private http: HttpClient) {
    }

    /** Create new facility site */
    create(facilitySite: FacilitySite): Observable<FacilitySite> {
        const url = `${this.baseUrl}`;
        return this.http.post<FacilitySite>(url, facilitySite);
    }

    /** GET specified facility site from the server */
    retrieve(id: number): Observable<FacilitySite> {
        const url = `${this.baseUrl}/${id}`;
        return this.http.get<FacilitySite>(url);
    }

    /** GET facility site by eis program ID and report ID */
    retrieveForReport(programId: string, reportId: number): Observable<FacilitySite> {
        const url = `${this.baseUrl}/report/${reportId}/facility/${programId}`;
        return this.http.get<FacilitySite>(url);
    }

    update(facilitySite: FacilitySite): Observable<FacilitySite> {
        const url = `${this.baseUrl}/${facilitySite.id}`;
        return this.http.put<FacilitySite>(url, facilitySite);
    }

    /** Create facility NAICS */
    createFacilityNaics(facilityNaics: FacilityNaicsCode): Observable<{}> {
        const url = `${this.baseUrl}/naics/`;
        return this.http.post<FacilityNaicsCode>(url, facilityNaics);
    }

    /** Update facility NAICS */
    updateFacilityNaics(facilityNaics: FacilityNaicsCode): Observable<FacilityNaicsCode> {
        const url = `${this.baseUrl}/naics/${facilityNaics.id}`;
        return this.http.put<FacilityNaicsCode>(url, facilityNaics);
    }

    /** DELETE facility NAICS from the database */
    deleteFacilityNaics(facilityNaicsId: number): Observable<{}> {
        const url = `${this.baseUrl}/naics/${facilityNaicsId}`;
        return this.http.delete(url);
    }
}
